File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/FilterDataViewModel.ts

export interface IFilterDataViewModel<T>  {
    data: T[];
    page: number;
    pageSize: number;
    totalRecord: number;
}
export class FilterDataViewModel<T>  implements IFilterDataViewModel<T> {
    data: T[];
    page: number;
    pageSize: number;
    totalRecord: number;
}

File: ProjectStart/ProjectStart.Common/ViewModel/FilterDataViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ProjectStart.Common.ViewModel
{
    public class FilterDataViewModel<T>
    {
        public IEnumerable<T> Data { get; set; }
        public int Page { get; set; }
        public int PageSize { get; set; }
        public int TotalRecord { get; set; }
    }
}
